### PR TITLE
#163221851 Users should be able to delete their articles

### PR DESCRIPTION
**Pull request by akhilome**
#### What does this PR do?

Add delete article functionality as well as implementing the protected routes functionality.

#### Description of Task to be completed?

- implement delete article functionality
- properly test all added code
- implement private route functionality

#### How should this be manually tested?

- clone the repo and `cd` into it
- run `npm install` to install the project dependencies
- rename `.env.sample` to `.env` and fill out the required details
- run `npm run start:dev` to start up the app in development mode
- navigate to the `/login` page to login to your account or `/signup` if you don't have one
- after sign up navigate to the `/create-article` page to create a new article which you can delete
- upon successful creation of an article, navigate to the `/profile/articles` page to view all your published articles
- interact with the page to delete the article created in one of the previous steps above

#### Any background context you want to provide?

N/A

#### What are the relevant pivotal tracker stories? (If applicable)

[#163221851](https://www.pivotaltracker.com/story/show/163221851), [#163467045](https://www.pivotaltracker.com/story/show/163467045)

#### Screenshots (If applicable)

screenshot 2019-02-03 at 6 54 01 pm

screenshot 2019-02-03 at 6 54 09 pm

screenshot 2019-02-03 at 6 54 15 pm

screenshot 2019-02-03 at 6 56 22 pm

screenshot 2019-02-03 at 6 56 39 pm

### CODE REVIEW

**Review by luqmanoop — APPROVED** [review]
Nice and clean implementation @akhilome. 👍

**Review by okoroemeka — APPROVED** [review]

**Review by omokaro-faith — APPROVED** [review]

**Comment by omokaro-faith** [comment]
@akhilome Plese resolve conflict.

File: src/components/auth/PrivateRoute.jsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { Route, Redirect } from 'react-router-dom';
+import { connect } from 'react-redux';
+import PreLoader from '../presentation/PreLoader';
+
+const PrivateRoute = ({ component: Component, isLoggedIn, ...rest }) => (
```
Comment (frostyblok) [inline src/components/auth/PrivateRoute.jsx]: This is a very good implementation, Kizito!

Comment (frostyblok) [inline src/components/auth/PrivateRoute.jsx:6]: This is a very good implementation, Kizito!

File: src/App.jsx [hunk 2]
```diff
@@ -29,13 +31,14 @@ class App extends Component {
...
           <Switch>
-            <Route exact path="/" component={Index} />
+            <Route exact path="/" component={Home} />
             <Route path="/login" component={Login} />
             <Route path="/signup" component={Signup} />
             <Route path="/create-article" component={CreateArticle} />
             <Route path="/articles/:slug" component={ViewArticle} />
             <Route path="/articles" component={AllArticles} />
-            <Route path="/profile" component={ProfilePage} />
+            <PrivateRoute path="/profile" component={ProfilePage} exact />
```
Comment (oluwajuwon) [inline src/App.jsx:40]: nice initiative implementing this PrivateRoute 👍, but Is there any reason why you didn't do this on the other routes that need to be protected?

**Review by frostyblok — APPROVED** [review]
This looks good to me!

**Review by oluwajuwon — COMMENTED** [review]
Nice job 👍 @akhilome.

I left a comment on your PR

**Comment by akhilome** [comment]
> @akhilome Plese resolve conflict.

Done!

**Pull request merged**